pinot-common/src/test/java/org/apache/pinot/common/metrics/MetricValueUtils.java

```diff
@@ -0,0 +1,128 @@
+    PinotMetric pinotMetric = extractMetric(metrics, metricName);
+    if (pinotMetric == null) {
+      return 0;
+    }
+    return ((YammerSettableGauge<Long>) pinotMetric.getMetric()).value();
+  }
+
+  public static boolean globalGaugeExists(AbstractMetrics metrics, AbstractMetrics.Gauge gauge) {
+    return extractMetric(metrics, gauge.getGaugeName()) != null;
+  }
+
+  public static long getGlobalGaugeValue(AbstractMetrics metrics, AbstractMetrics.Gauge gauge) {
+    PinotMetric pinotMetric = extractMetric(metrics, gauge.getGaugeName());
+    if (pinotMetric == null) {
+      return 0;
+    }
+    return ((YammerSettableGauge<Long>) pinotMetric.getMetric()).value();
+  }
+
+  public static boolean globalGaugeExists(AbstractMetrics metrics, String key, AbstractMetrics.Gauge gauge) {
+    return extractMetric(metrics, gauge.getGaugeName() + "." + key) != null;
+  }
+
+  public static long getGlobalGaugeValue(AbstractMetrics metrics, String key, AbstractMetrics.Gauge gauge) {
+    PinotMetric pinotMetric = extractMetric(metrics, gauge.getGaugeName() + "." + key);
+    if (pinotMetric == null) {
+      return 0;
+    }
+    return ((YammerSettableGauge<Long>) pinotMetric.getMetric()).value();
+  }
+
+  public static boolean tableGaugeExists(AbstractMetrics metrics, String tableName, AbstractMetrics.Gauge gauge) {
+    return extractMetric(metrics, gauge.getGaugeName() + "." + tableName) != null;
+  }
+
+  public static long getTableGaugeValue(AbstractMetrics metrics, String tableName, AbstractMetrics.Gauge gauge) {
+    PinotMetric pinotMetric = extractMetric(metrics, gauge.getGaugeName() + "." + tableName);
+    if (pinotMetric == null) {
+      return 0;
+    }
+    return ((YammerSettableGauge<Long>) pinotMetric.getMetric()).value();
+  }
+
+  public static boolean tableGaugeExists(AbstractMetrics metrics, String tableName, String key,
+      AbstractMetrics.Gauge gauge) {
+    return extractMetric(metrics, gauge.getGaugeName() + "." + tableName + "." + key) != null;
+  }
+
+  public static long getTableGaugeValue(AbstractMetrics metrics, String tableName, String key,
+      AbstractMetrics.Gauge gauge) {
+    PinotMetric pinotMetric = extractMetric(metrics, gauge.getGaugeName() + "." + tableName + "." + key);
+    if (pinotMetric == null) {
+      return 0;
+    }
+    return ((YammerSettableGauge<Long>) pinotMetric.getMetric()).value();
+  }
+
+  public static boolean partitionGaugeExists(AbstractMetrics metrics, String tableName, int partitionId,
+      AbstractMetrics.Gauge gauge) {
+    return extractMetric(metrics, gauge.getGaugeName() + "." + tableName + "." + partitionId) != null;
+  }
+
+  public static long getPartitionGaugeValue(AbstractMetrics metrics, String tableName, int partitionId,
+      AbstractMetrics.Gauge gauge) {
+    PinotMetric pinotMetric = extractMetric(metrics, gauge.getGaugeName() + "." + tableName + "." + partitionId);
+    if (pinotMetric == null) {
```

Comment:
The previous implementation for `getValueOfPartitionGauge()` returned `-1`. Should we keep the backward compatibility..?

Reply:
This `getValueOfPartitionGauge` is only used in unit tests and integration tests, as long as we don't break those tests, it should be fine even if we don't return `-1` in the new method

Reply:
The code is defined under `src/test` path, so it's by default visible to tests only